### PR TITLE
BLD: upgrade salad config for multiple nodes

### PR DESCRIPTION
**Pull request by lacabra**
This PR introduces the necessary updates to align the network configuration in salad to support the latest version of `discovery-cli` which allows for running a docker network with multiple nodes, in the following manner (for example with 3 worker nodes):
```
NODES=3 yarn dc start
```

Significant changes:
- Addition of a new container: a `nginx` instance that is configured as a load balancer to distribute the JSON-RPC requests to any P2P node in the network. Thus the external port is changed from `3346` (which remains internal) to `3333`.  Added `conf/nginx.conf` that configures this container at boot time.
- changed `.env.template` to account for the above change
- pulled new docker-compose files from https://github.com/enigmampc/discovery-docker-network/tree/develop/config (while maintaining the `mongo` stanza). Best practice now is to soft-link docker-compose.yml to either one of them.
- updated `migrations/2_deploy_contracts.js` to take into account environment variables from `.env`

All tests passing locally, ready to merge.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 .env
 .idea
 node_modules
-etc
\ No newline at end of file
+etc
+build/
```
Comment (fredfortier) [inline .gitignore]: @lacabra Can you confirm that `discovery-cli` no longer requires an existing `build/enigma_contracts/' folder?

Comment (lacabra) [reply .gitignore]: Great observation. No that is not the case, and that folder is still required, so I have adjusted `.gitignore` accordingly in the time being in ac56a65

File: docker-compose.cli-sw.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+version: '3'
+
+services:
+
+  contract:
+    image: enigmampc/enigma_contract:${DOCKER_TAG}
```
[the remark this answers is not in the document]
Comment (lacabra) [reply docker-compose.cli-sw.yml]: We discussed it offline that the choice of a makefile vs scripting is a matter of personal choice, so leaving this up for another time

File: config/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+	client_max_body_size 20M;
+
+    server {
+          listen 3333;
+          location / {
+            proxy_pass http://p2p:3346;
+          }
+    }
+}
```
[the remark this answers is not in the document]
Comment (lacabra) [reply config/nginx.conf]: Yes, there was an issue with my editor, I noticed it later. Fixed it in 6f9a4f5

**Pull request merged**